rllib/algorithms/ppo/tests/test_ppo_with_rl_module.py

```diff
@@ -1,18 +1,93 @@
-import numpy as np
 import unittest
 
+import numpy as np
+import tree
+import gym
+import torch
+
 import ray
 import ray.rllib.algorithms.ppo as ppo
-
 from ray.rllib.algorithms.callbacks import DefaultCallbacks
+from ray.rllib.algorithms.ppo.torch.ppo_torch_rl_module import (
+    PPOModuleConfig,
+    PPOTorchRLModule,
+    get_ppo_loss,
+)
+from ray.rllib.core.rl_module.encoder import (
+    IdentityConfig,
+    FCConfig,
+    LSTMConfig,
+)
 from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+from ray.rllib.policy.sample_batch import SampleBatch
 from ray.rllib.utils.metrics.learner_info import LEARNER_INFO, LEARNER_STATS_KEY
+from ray.rllib.utils.numpy import convert_to_numpy
 from ray.rllib.utils.test_utils import (
     check,
     check_compute_single_action,
     check_train_results,
     framework_iterator,
 )
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+
+
+# Get a model config that we expect from the catalog
+def get_expected_model_config(env, lstm, shared_encoder):
```

Comment:
add type hints and a little bit of docstring to remind what it does. The function has got a bit complicated and deserves a little documentation, despite the possibility of getting removed in favor of a catalog stuff soon.

Reply:
I'm adding some stuff here and will also extend this method a little.
We'll settle on Model APIs and other conventions soon, and that'll be the point where I'll ditch this method in favour of the catalog.